lib/cards/KCardGrid.vue

```diff
@@ -16,28 +55,40 @@
   import { watch, ref, provide } from '@vue/composition-api';
 
   import { LAYOUT_1_1_1, LAYOUT_1_2_2, LAYOUT_1_2_3 } from './gridBaseLayouts';
-  import useResponsiveGridLayout from './useResponsiveGridLayout';
+  import useGridLayout from './useGridLayout';
+  import useGridLoading from './useGridLoading';
+  import SkeletonCard from './SkeletonCard';
 
   /**
    * Displays a grid of cards `KCard`.
-   *
-   * Offers default behavior corresponding to the most
-   * commonly used grids, as well as advance configuration
-   * via `useKCardGrid` to customize a base grid layout
-   * or even completely override it.
    */
   export default {
     name: 'KCardGrid',
-
+    components: {
+      SkeletonCard,
+    },
     setup(props) {
-      const { currentLevelConfig } = useResponsiveGridLayout(props);
+      const { currentBreakpointConfig, windowBreakpoint } = useGridLayout(props);
+      const {
+        finishedMounting,
+        isLoading,
+        skeletonCount,
+        skeletonHeight,
+        skeletonOrientation,
+        skeletonThumbnailDisplay,
+        skeletonThumbnailAlign,
+      } = useGridLoading(props);
 
       const gridStyle = ref({});
       const gridItemStyle = ref({});
 
       watch(
-        currentLevelConfig,
+        currentBreakpointConfig,
```

Comment:
Just out of curiosity - what is the reason we're doing a watch and assign here, rather than a computed prop? If there's a specific reason, might be good to have a code comment accordingly, like we do in other places where we use a similar pattern.

Reply:
Thank you, ah yes - I admit I forgot the reason for this particular place. I think it was related to general problem I had - I couldn't make the `computed` approaches work in the accompanying composables. I was able to track it to `useKResponsiveWindow`, but am still not clear what exactly was going on. Was planning to debug and discuss it later together with the unit testing issue, would that be okay?

Reply:
Opened issue https://github.com/learningequality/kolibri-design-system/issues/801